solo-chains/runtime/dancelight/src/lib.rs

```diff
@@ -1618,11 +1620,54 @@ impl pallet_data_preservers::AssignmentPayment<AccountId> for PreserversAssignme
     }
 }
 
+pub struct DataPreserversBenchmarkHelper<T>(PhantomData<T>);
+
+#[cfg(feature = "runtime-benchmarks")]
+impl<T> ArgumentFactory<T::AccountId> for DataPreserversBenchmarkHelper<T>
+where
+    T: pallet_registrar::Config + pallet_balances::Config + paras_registrar::Config,
+{
+    fn reserve_para_id(para_id: ParaId) {
+        use frame_support::{assert_ok, dispatch::RawOrigin, traits::fungible::Mutate};
+
+        let account: T::AccountId = frame_benchmarking::account("para_manager", 0, 1);
+        assert_ok!(<T as pallet_registrar::Config>::Currency::mint_into(
```

Comment:
Here you can probably call `force_register` instead of register. This should avoid:
- sending tokens to the register account
- need to reserve the para-id (maybe I am wrong here).
- all the nextFreeParaId stuff.
-

Reply:
I tried using `paras_registrar::force_register` however  I either get 'NotOwner' error if I don't use `pallet_registrar::register`  after or `AlreadyRegistered `error if I do call it

Reply:
Moved reserving ParaId to `InnerDancelightRegister::prepare_chain_registration ` to avoid duplicating code in https://github.com/moondance-labs/tanssi/pull/783/commits/674169290cfa373cac5047de830553dc0e85d3d3